mailpoet/mailpoet
# Remove some PHPStan Level 6 errors - 2

Testing instructions for GlobalStep:
1. Go to MailPoet > Lists > Segment tab
2. Please perform some testing around segments to ensure there are no issues

 ---- 

This continues the work from https://github.com/mailpoet/mailpoet/pull/4138
It can be merged separately.

This PR removes some of the following PHPStan level 6 errors from MailPoet\Segments:

To test you can modify the level6 exclusion rule on phpstan.neon file to this:

    # exclude level 6 errors (but keep them for Automation)
    - '/(Method|Property|Function) (?!(MailPoet\\Automation\\)|(MailPoet\\Segments\\)|(MailPoet\\Test\\Segments\\)).*has no (return )?type specified/'
    - '/(Method|Function) (?!(MailPoet\\Automation\\)|(MailPoet\\Segments\\)|(MailPoet\\Test\\Segments\\)).*has parameter (\$[_A-Z]{1}[_a-z]+)? with no type (specified)?/i'


run `./do qa:phpstan` and check that the committed files throw no errors.

[MAILPOET-3720](https://mailpoet.atlassian.net/browse/MAILPOET-3720?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ)

There are too many errors to fix in a couple of PRs. I will continue working on this and try to fix as many as possible before my sabbatical :)
